MainMovement.py

```diff
@@ -0,0 +1,163 @@
+        if len(processedData.balls) > 0:
+            #print(processedData.balls)
+            return States.drive
+        else:
+            # otherwise it spins until it finds some
+            self.motion.move(0, 0, 1, 0)
+        return States.spin
+
+    def drive(self, processedData):
+        y_keskmine = list()
+        if len(processedData.balls) > 0:
+            ball = processedData.balls[-1]
+            #print("distance: {}".format(ball))
+            xmiddif = (424 - (ball.x)) * 0.002              #approcing ball relative to the ball's x and y coordinate
+            yfor = (360 - (ball.distance)) * 0.005
+            #print("x: ", xmiddif, ", y: ", yfor)
+            self.motion.move(0, yfor, xmiddif, 0)
+            if ball.distance >= 350:
+                # if it is close enough to a ball, it should start orbiting around it to find the basket:
+                return States.orbit
+        # if it loses the ball for some reason (opponent takes it), it starts spinning to look for a new one:
+        else:
+            return States.spin
+        return States.drive
+
+    def orbit(self, processedData, basketColor):
+        
+        basket = processedData.basket_b if basketColor == 'b' else processedData.basket_m
+        
+        frame_centre_x = 848/2
+       
+
+        if len(processedData.balls) > 0 : #checks if robot can see ball
+            ball = processedData.balls[-1]
+            orbitrad = (400 - ball.distance) *0.003
+            orbitvar = (frame_centre_x - ball.x )*0.009 #variable
+            
+            orbitdirmultiplier = 1 #used to assign the direction of orbit
+           
+            pixel_difference = 2
+            # if it has a basket in frame and it is centred, it goes to throw the ball:
+            if basket.exists and frame_centre_x -pixel_difference  < basket.x < frame_centre_x + pixel_difference:
+
+                print("could throw")
+                return States.throw
+            # if it has a ball and a basket in frame, but the basket isn't centred:
+            elif basket.exists:
+                orbitdirmultiplier = (frame_centre_x - basket.x)*0.002
+                #print(orbitdirmultiplier)
+                if orbitdirmultiplier > 0.1:
+                    orbitdirmultiplier = 0.1
+                elif orbitdirmultiplier < -0.1:
+                    orbitdirmultiplier = -0.1
+
+
+                self.motion.move(orbitdirmultiplier, orbitrad, orbitvar, 0)
+                # centre basket and ball to frame
+            # if it can't find a basket in the frame, it orbits faster:
+            elif not basket.exists:
+
+                self.motion.move(0.3, orbitrad, orbitvar, 0)
+        # if it lost the ball, it goes to look for a new one:
+        else:
+            return States.spin
+        return States.orbit
+
+
+    def throw(self, processedData,basketColor,basketdistcm):
+
+        basket = processedData.basket_b if basketColor == 'b' else processedData.basket_m
```

Comment:
Random string here as well

Reply:
fixed this with color enums from color.py